uint64 inc
uint8 direction
uint8 velocity
uint8 rotation